lib/contacts/distributed_lock.rb

```diff
@@ -0,0 +1,29 @@
+require 'redis'
+require 'redis-lock'
+
+module Contacts
+  class DistributedLock
+    LIFETIME = (5 * 60) # seconds. The lock expires automatically after this time
+
+    def initialize(lock_name)
+      @lock_name = lock_name
+    end
+
+    def lock
+      redis.lock("contacts-admin:#{Rails.env}:#{@lock_name}", life: LIFETIME) do
```

Comment:
If we're specifying the namespace in the config file, we should avoid hard-coding it here.